Define 'resource' in glossary (#49706)
kubernetes/website: content/en/docs/reference/glossary/infrastructure-resource.md

---
title: Resource (infrastructure)
id: infrastructure-resource
date: 2025-02-09
short_description: >
  A defined amount of infrastructure available for consumption (CPU, memory, etc).

aka:
tags:
- fundamental
---

> Capabilities provided by {{< glossary_tooltip text="nodes" term_id="node" >}} (CPU, memory, GPUs, etc), and made available for consumption by processes or users of those nodes.
> <br>

Reviewer: We can avoid HTML here:
```suggestion
Capabilities provided to one or more {{< glossary_tooltip text="nodes" term_id="node" >}} (CPU, memory, GPUs, etc), and made available for consumption by
{{< glossary_tooltip text="Pods" term_id="pod" >}} running on those nodes.

```

Author: Note: the rendered text doesn't have a line break

Kubernetes also uses the term _resource_ to describe {{< glossary_tooltip text="API resource" term_id="api-resource" >}}.

<!--more-->

> Computers provide fundamenal capabilties: processing power, storage memory, network, etc

Reviewer: ```suggestion
Computers provide fundamental hardware facilities: processing power, storage memory, network, etc.
```

> These resources have finite capacity, measured in a unit applicable to that resource (Number of CPUs, bytes of memory, etc).

Reviewer: ```suggestion
These resources have finite capacity, measured in a unit applicable to that resource (number of CPUs, bytes of memory, etc).
```

> Kubernetes abstracts common resources for allocation to workloads.
> And utilizes operating system primatives (e.g. {{< glossary_tooltip text="cgroups" term_id="cgroup" >}}) to manage consumption by {{< glossary_tooltip text="workloads" term_id="workload" >}}.

Reviewer: ```suggestion
Kubernetes abstracts common [resources](/docs/concepts/configuration/manage-resources-containers/)
for allocation to workloads and utilizes operating system primitives (for example, Linux {{< glossary_tooltip text="cgroups" term_id="cgroup" >}}) to manage consumption by {{< glossary_tooltip text="workloads" term_id="workload" >}}).

You can use [dynamic resource allocation](/docs/concepts/scheduling-eviction/dynamic-resource-allocation/) to
manage complex resource allocations automatically.
```

---
title: Resource (infrastructure)
id: infrastructure-resource
date: 2025-02-09
short_description: >
  A defined amount of infrastructure available for consumption (CPU, memory, etc).

aka:

> - architecture

Author: ```suggestion
tags:
- architecture
```

---
title: Resource (infrastructure)
id: infrastructure-resource
date: 2025-02-09
short_description: >
  A defined amount of infrastructure available for consumption (CPU, memory, etc).

aka:

> tags:
> - fundamental

Reviewer: (IMO)
```suggestion
- architecture
```

Author: (fixed the removed `tags:` in commit below)

---
Capabilities provided to one or more {{< glossary_tooltip text="nodes" term_id="node" >}} (CPU, memory, GPUs, etc), and made available for consumption by
{{< glossary_tooltip text="Pods" term_id="pod" >}} running on those nodes.

> Kubernetes also uses the term _resource_ to describe {{< glossary_tooltip text="API resource" term_id="api-resource" >}}.

Author: ```suggestion
Kubernetes also uses the term _resource_ to describe an {{< glossary_tooltip text="API resource" term_id="api-resource" >}}.
```

Author: (also see below)

> Kubernetes also uses the term _resource_ to describe {{< glossary_tooltip text="API resource" term_id="api-resource" >}}.

Author: ```suggestion
Kubernetes also uses the term _resource_ to describe {{< glossary_tooltip text="API resources" term_id="api-resource" >}}.
```

Author: not sure which one of these reads better

<!--more-->
Computers provide fundamental hardware facilities: processing power, storage memory, network, etc.
These resources have finite capacity, measured in a unit applicable to that resource (number of CPUs, bytes of memory, etc).
Kubernetes abstracts common [resources](/docs/concepts/configuration/manage-resources-containers/)
for allocation to workloads and utilizes operating system primitives (for example, Linux {{< glossary_tooltip text="cgroups" term_id="cgroup" >}}) to manage consumption by {{< glossary_tooltip text="workloads" term_id="workload" >}}).

> You can use [dynamic resource allocation](/docs/concepts/scheduling-eviction/dynamic-resource-allocation/) to

Reviewer: Sure
